engine-standalone-storage/src/relayer_db/types.rs

```diff
@@ -68,35 +69,81 @@ pub struct BlockRow {
     pub receipts_root: H256,
 }
 
-impl From<postgres::Row> for BlockRow {
-    fn from(row: postgres::Row) -> Self {
-        let chain: i32 = row.get("chain");
-        let id: i64 = row.get("id");
+struct BlockRowSize(i32);
+
+impl TryFrom<BlockRowSize> for u32 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: BlockRowSize) -> Result<Self, Self::Error> {
+        // set negative values to 0
+        value.0.max(0).try_into()
+    }
+}
+
+struct BlockRowChain(i32);
+
+impl TryFrom<BlockRowChain> for u64 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: BlockRowChain) -> Result<Self, Self::Error> {
+        // set negative values to 0
+        value.0.max(0).try_into()
+    }
+}
+
+struct BlockRowId(i64);
+
+impl TryFrom<BlockRowId> for u64 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: BlockRowId) -> Result<Self, Self::Error> {
+        // set negative values to 0
+        value.0.max(0).try_into()
+    }
+}
+
+struct BlockRowBlock(i64);
+
+impl TryFrom<BlockRowBlock> for u64 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: BlockRowBlock) -> Result<Self, Self::Error> {
+        // set negative values to 0
+        value.0.max(0).try_into()
+    }
+}
+
+impl TryFrom<postgres::Row> for BlockRow {
+    type Error = postgres::Error;
```

Comment:
```suggestion
    type Error = TryFromIntError;
```

Reply:
But then trait bounds are not met for the type as super::initialize_blocks function in mod.rs requires postgres::Error to return. I tried to return both Error (TryFromIntError and postgres::Error) with an enum type, but iterating RowIter type always requires only postgres::Error to return. Hence, I had to include the integer conversion error into postgres::Error type.